src/libpst.c

```diff
@@ -3564,8 +3622,13 @@ static int pst_getBlockOffsetPointer(pst_file *pf, pst_id2_tree *i2_head, pst_su
         }
     }
     else {
+        DEBUG_WARN(("Found internal %#x value.\n", offset));
         // internal index reference
         size_t subindex  = offset >> 16;
+        if (pf->do_read64 == 2) {
+            // Shift over 3 more bits for new flags.
+            subindex = subindex >> 3;
```

Comment:
libpst code is confusing in general because it had to be written without the spec. libpst 0.1 came out in 2002. The first public release of the PST spec wasn't until 2010!

* `offset` in this function is an HNID (HID/NID union).
* `if ((offset & 0xf) == (uint32_t)0xf)` distinguishes whether the `hidType` is 0 (NID_TYPE_HID) or 0x1F (NID_TYPE_LTP). It should really be checking the bottom 5 bits, not 4, but in practice I don't think HNIDs are used to point to any other type of NID besides NID_TYPE_LTP, so it wouldn't make any difference.
* In this else block, we know `offset` is a HID. Per the spec, the `hidIndex` field in bits 5-15 is an index into an array of 2-byte offsets. libpst instead uses bits *4*-15 as a *byte offset*. It "works" because bits 0-4 are always zero in a HID.